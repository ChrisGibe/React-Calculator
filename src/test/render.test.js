import renderer from 'react-test-renderer';
import Button from '../components/Button'
import Display from '../components/Display'
import React from 'react';

describe('check the rendering of components', () => {
  it('Diplay component renders correctly', () => {
    const tree = renderer
    .create(<Display readOnly={true} value="23" />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('Display component renders correctly 2', () => {
    const tree = renderer
      .create(<Display readOnly={true} value="{ButtonClick.clickEvent('2')}" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Display component renders correctly C', () => {
    const tree = renderer
      .create(<Display readOnly={true} value="{ButtonClick.clickEvent('C')}" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button component renders correctly', () => {
    const tree = renderer
    .create(<Button label={{"label" : 4}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
