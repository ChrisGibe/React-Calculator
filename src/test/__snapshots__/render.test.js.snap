// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the rendering of components Button component renders correctly 1`] = `
<input
  className="calc-btn"
  onClick={[Function]}
  type="button"
  value={
    Object {
      "label": 4,
    }
  }
/>
`;

exports[`check the rendering of components Diplay component renders correctly 1`] = `
<textarea
  className="calc-display"
  value="23"
/>
`;

exports[`check the rendering of components Display component renders correctly 2 1`] = `
<textarea
  className="calc-display"
  value="{ButtonClick.clickEvent('2')}"
/>
`;

exports[`check the rendering of components Display component renders correctly C 1`] = `
<textarea
  className="calc-display"
  value="{ButtonClick.clickEvent('C')}"
/>
`;
